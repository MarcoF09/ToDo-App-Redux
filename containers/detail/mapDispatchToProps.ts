import { Dispatch } from 'react';
import { MapDispatchToProps } from 'react-redux';
import { Actions } from '../../actions/actions';
import { State, Todo } from '../../types/globalTypes';
import { Actions as ActionCreator } from '../../types/types';

export const mapDispatchToProps: MapDispatchToProps<
  Dispatch<ActionCreator>,
  State
> = dispatch => ({
  markAsNotDone: (todos: Todo[], index: number) => {
    dispatch(Actions.markAsNotDone(todos, index));
  },
  markAsDone: (todos: Todo[], index: number) => {
    dispatch(Actions.markAsDone(todos, index));
  }
});
