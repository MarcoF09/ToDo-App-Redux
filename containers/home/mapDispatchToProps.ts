import { Dispatch } from 'react';
import { MapDispatchToProps } from 'react-redux';
import { Actions } from '../../actions/actions';
import { State, Todo } from '../../types/globalTypes';
import { Actions as ActionCreator } from '../../types/types';

export const mapDispatchToProps: MapDispatchToProps<
  Dispatch<ActionCreator>,
  State
> = dispatch => ({
  getToDoData: () => {
    dispatch(Actions.getToDoData());
  },
  clearAllDone: (todos: Todo[]) => {
    dispatch(Actions.clearAllDone(todos));
  },
  changeCheckBoxState: (todos: Todo[], index: number) => {
    dispatch(Actions.changeCheckBoxState(todos, index));
  },
  deleteItem: (todos: Todo[], index: number) => {
    dispatch(Actions.deleteItem(todos, index));
  }
});
