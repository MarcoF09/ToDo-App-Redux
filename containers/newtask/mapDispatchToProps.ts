import { Action } from 'redux'
import { ThunkDispatch } from 'redux-thunk'
import { Actions } from '../../actions/actions'
import { State } from '../../types/globalTypes'

export const mapDispatchToProps = (
  dispatch: ThunkDispatch<State, {}, Action>
) => ({
  addToDo: (title: string, description: string) => {
    dispatch(Actions.addToDo(title, description))
  }
})
