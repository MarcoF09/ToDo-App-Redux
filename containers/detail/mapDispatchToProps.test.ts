import { mapDispatchToProps } from './mapDispatchToProps'

describe('should return two method to handle if a todo is done or not', () => {
  const dispatch = jest.fn()
  it('methods returned correctly', () => {
    expect(mapDispatchToProps(dispatch)).toMatchSnapshot()
  })
})
