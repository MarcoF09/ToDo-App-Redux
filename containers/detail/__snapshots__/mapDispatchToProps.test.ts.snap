// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return two method to handle if a todo is done or not methods returned correctly 1`] = `
Object {
  "markAsDone": [Function],
  "markAsNotDone": [Function],
}
`;
